lib/elastic_apm/context.rb

```diff
@@ -9,13 +9,24 @@
 module ElasticAPM
   # @api private
   class Context
-    attr_accessor :request, :response, :user
-    attr_reader :custom, :tags
-
     def initialize(custom: {}, tags: {}, user: nil)
       @custom = custom
       @tags = tags
       @user = user || User.new
     end
+
+    attr_accessor :request
+    attr_accessor :response
+    attr_accessor :user
+    attr_reader :custom
+    attr_reader :tags
+
+    def empty?
+      return false if tags.any?
+      return false if custom.any?
+      return false unless user.empty?
+
+      true
```

Comment:
You can sum up the last check to have:
```
def empty?
      return false if tags.any?
      return false if custom.any?
      user.empty?
end
```

Reply:
Right, I thought about that but liked the explicitness of this approach :)

Reply:
ok, that's fine with me.